src/ol/events/eventtarget.js

```diff
@@ -64,12 +71,17 @@ ol.events.EventTarget.prototype.dispatchEvent = function(event) {
   evt.target = this;
   var listeners = this.listeners_[type];
   if (listeners) {
-    for (var i = listeners.length - 1; i >= 0; --i) {
-      if (listeners[i].call(this, evt) === false ||
-          evt.propagationStopped) {
+    this.pendingRemovals_[type] = 0;
```

Comment:
This is a narrow case that we may decide is not worth supporting, but if a listener dispatches another of the same event, this will incorrectly be set to `0` (if previous listeners have been removed).